test/dom-parsing/XMLSerializer.tests.ts

```diff
@@ -283,6 +283,12 @@ describe('XMLSerializer', () => {
 		);
 	});
 
+	it('escapes whitespace that would otherwise be lost due to attribute value normalization', () => {
+		const el = document.createElement('test');
+		el.setAttribute('whitespace', ' \t\r\n');
+		expect(serializer.serializeToString(el)).toBe('<test whitespace=" &#9;&#13;&#10;"/>');
```

Comment:
Could you add a test for double-escapes? el.setAttribute('whitespace', " &#9;&#13;&#10;");